### PR TITLE
RLP-801 Fixing settlement light

### PR DESCRIPTION
**Pull request by JONAF2103**
This fixes an error caused by the string comparison, now we use error codes, also fixes the from_message counter that was being reseted when the list of messages was empty.

### CODE REVIEW

File: src/store/actions/settle.js [hunk 1]
```diff
@@ -37,20 +37,14 @@ export const settleChannel = data => async (dispatch, _, lh) => {
...
-    const alreadyUnlockErr = "Channel is already unlocked.";
-    const alreadySettledErr = "channel that's already settled";
-    const alreadySignedErr = "Message already signed";
     const requestError = error?.response?.data?.errors;
     if (requestError) {
-      const wasUnlocked = requestError?.includes(alreadyUnlockErr);
-      const wasSettled = requestError?.includes(alreadySettledErr);
-      const alreadySigned = requestError?.includes(alreadySignedErr);
-      if (wasSettled || wasUnlocked || alreadySigned) {
+      const channelAlreadySettled = requestError === "CHANNEL_ALREADY_SETTLED" || requestError === "MESSAGE_ALREADY_SIGNED";
```
Comment (marcosmartinez7) [inline src/store/actions/settle.js:42]: I think this is ok but what about the "Channel is already unlocked." check? why it is not required now?

**Review by marcosmartinez7 — COMMENTED** [review]
lgtm

File: src/store/actions/settle.js (hunk 1 above)
Comment (JONAF2103) [reply src/store/actions/settle.js:42]: I forgot that and i found it failing, i fixed that now and it's in the new commit that i'm going to upload

File: src/store/sagas/index.js [hunk 2]
```diff
@@ -54,9 +54,10 @@ export function* workMessagePolling({ data }) {
     });
     yield all(completed.map(paymentId => setCompleted(paymentId)));
     const maxIdentifier = findMaxMsgInternalId(data);
-
-    yield put({ type: SET_LATEST_INTERNAL_MSG_ID, id: maxIdentifier });
-
+    if (data && data.length && data.length >= 0) {
```
Comment (mauricioirace) [inline src/store/sagas/index.js:57]: I don't understand the last term of the conjunction. Isn't data.length always non-negative?

Comment (JONAF2103) [reply src/store/sagas/index.js:57]: don't understand your question

Comment (JONAF2103) [reply src/store/sagas/index.js:57]: we are checking if data.length is at least 1 or more

**Review by marcosmartinez7 — APPROVED** [review]
lgtm now!

File: src/store/sagas/index.js (hunk 2 above)
Comment (mauricioirace) [reply src/store/sagas/index.js:57]: you are checking if data.length is 0 or more

Comment (JONAF2103) [reply src/store/sagas/index.js:57]: true i will fix that

**Review by mauricioirace — APPROVED** [review]
LGTM!!

**Pull request merged**